docs/dev/guides/bootstrap.md

```diff
@@ -0,0 +1,182 @@
+
+```bash
+# Clone the pulp-cli repository if you don't have it already
+git clone https://github.com/pulp/pulp-cli.git
+
+# Run the bootstrap script
+pulp-cli/cookiecutter/apply_templates.py --bootstrap
+
+```
+
+### Answer the Prompts
+
+You will be prompted for several values during the bootstrap process. The most important is **app_label**, which by convention (when applicable) should match the Pulp server component your plugin targets (for example, use `file` for `pulp-file`). This value determines the import paths and command names for your plugin. You will also see additional prompts for options such as glue integration, documentation, translations, version, repository URL, and CI configuration—answer these as appropriate for your project.
+
+### Key Files and Directories
+
+The bootstrap process creates several important directories and files:
+
+- `pulpcore/cli/my-plugin/__init__.py`: Main entry point where you define command groups. See bellow for more details
+- `pulp-glue-my-plugin/pulp_glue/my-plugin/context.py`: API context class for interacting with Pulp. See bellow for more details
+- `pyproject.toml`: Project metadata, dependencies, and build configuration. This file contains essential project information including author details, licensing information, version numbers, package dependencies, and build system configuration. You'll need to customize the author name, email, project description, license type to match your plugin's specific requirements.
+- `CHANGES/`: Directory for changelog fragments (.feature, .bugfix, etc. files). See the [changelog update guide](https://pulpproject.org/pulpcore/docs/dev/guides/git/?h=towncrier#changelog-update) for more details.
+- `.github/workflows/`: This directory contains GitHub Actions workflow configurations for automated testing and continuous integration. These files are pre-configured and do not require modification when initially setting up your plugin development environment. The CI workflows will automatically run tests and validation checks on your plugin code when you push changes to your repository.
+
+---
+
+## Customizing Your Plugin
+
+### Create API Context Classes
+
+Edit `pulp-glue-my-plugin/pulp_glue/my_plugin/context.py` to define context classes that handle API interactions:
+
+```python
+import typing as t
+from gettext import gettext as _
+
+from pulp_glue.common.context import PulpContext, PulpEntityContext
+
+
+class PulpMyExampleContext():
+    """Context for working with my custom resource."""
+
+    ID_PREFIX = "my_resource"
+
+    def example_action(self, data: t.Dict[str, t.Any]) -> t.Dict[str, t.Any]:
+        """Create a custom resource with specific data.
+
+        Args:
+            data: The data dictionary to send to the API
+
+        Returns:
+            The created resource entity
+        """
+        response = self.call(
+            operation_id="my_resource_example_action",
+            body=data,
+            validate_body=False,
```

Comment:
```suggestion
```
That's meant to be a last resort. Whenever we can we want to encourage linting on the client side before bothering the server.